tests/test_numpy_array.py

```diff
@@ -477,6 +477,37 @@ def test_array_create_and_resize(msg):
     assert np.all(a == 42.0)
 
 
+def test_array_reshape(msg):
+    a = np.random.randn(10 * 10 * 10).astype("float64")
+    x = m.array_reshape1(a, 10)
+    assert x.shape == (10, 10, 10)
+
+
+@pytest.mark.xfail("env.PYPY")
+def test_create_and_reshape(msg):
+    x = m.create_and_reshape(10, 20, 30)
+    assert x.shape == (10, 20, 30)
+
+
+def test_reshape_tuple(msg):
+    a = np.random.randn(10, 20, 30).astype("float64")
+    x = m.reshape_tuple(a, (30, 20, 10))
+    assert x.shape == (30, 20, 10)
+
+
+def test_reshape_tuple_empty(msg):
+    a = np.random.randn(10 * 20 * 30).astype("float64")
+    # FIXME: This should be a ValueError for maximum numpy compat
```

Comment:
Yep, I just spent a couple minutes playing myself, we definitely need to make this compatible in this PR. Otherwise we'd be introducing a wrinkle of the really silly kind that can lead to surprisingly huge stumbles.

If you feel out of your comfort zone, I could help out working on it here. Please let me know.

Reply:
Yeah, just not sure how to fix the caster code here to allow empty iterable to be coercible into a std::vector<int> if it's empty. Could definitely use some help here, that seems like it might even warrant it's own PR, right?

Reply:
I get the following error back: 
```
E           TypeError: Unable to convert function return value to a Python type! The signature was
E           	(arg0: numpy.ndarray[numpy.float64], arg1: List[int]) -> numpy.ndarray
```

Reply:
@rwgk Aha, figured it out. Behavior matches Numpy now.